### ISSUE TITLE
HA: convert Desktop to utilize Caching API and make it stateless

### ISSUE THREAD

**Issue by jackjia-ibm** [zowe/zowe-install-packaging#1699]
*Customer content*

**Is your feature request related to a problem? Please describe.**

Currently Desktop stores some state data in memory which makes it stateful, we will need to convert it to stateless.

**Describe the solution you'd like**

After Caching API is implemented, we will need to convert Desktop to be stateless by storing the state data into Caching API.

**Describe alternatives you've considered**

If Caching Api is not defined and presented, the state data will be stored locally within the instance to avoid failures.

**Comment by MarkAckert** [comment]
FYI @1000TurquoisePogs and @NolanRogers , depends on #863

**Comment by 1000TurquoisePogs** [comment]
App-server tasks
Must do: https://github.com/zowe/zlux/issues/518
May be able to postpone: https://github.com/zowe/zlux/issues/519

**Comment by 1000TurquoisePogs** [comment]
zss tasks
https://github.com/zowe/zlux/issues/520
